app/models/session.rb

```diff
@@ -5,6 +5,13 @@ class Session < ApplicationRecord
   scope :alive, -> { where alive? }
   scope :expired, -> { where expired? }
 
+  def active?(providers = {})
+    return true unless self.expires_at
+    return false unless self.expires_at > Time.now.utc
+    return true if providers.empty?
+    providers.any? { |provider| self.data['provider'] == provider }
```

Comment:
Hi @pachonk ,
Thanks for your work on this. So, to clarify, line 12 is checking that the provider linked to the object is the same as the provider passed to the `active?` method? Why is the method returning true if no providers are passed to the method? Let me know if I'm misunderstanding something.